eventloop/eventloop.go

```diff
@@ -27,15 +27,32 @@ func GetGlobalEventLoop() *EventLoop {
 func (e *EventLoop) Await(currentP *Promise) interface{} {
 	defer currentP.Done()
 	currentP.RegisterHandler()
-	val := <-currentP.rev
-	return val
+	for {
+		select {
+		case err := <-currentP.errChan:
+			return err
+		case rev := <-currentP.rev:
+			return rev
+		}
+	}
+
 }
```

Comment:
I would prefer we return interface{} and error here, returning value and error as interface{} won't be handy to the users. what do you think?

Reply:
Done.